src/cogent3/core/new_alignment.py

```diff
@@ -3096,174 +3315,881 @@ def gapped_bytes_value(self) -> bytes:
+            .astype(numpy.uint8)
+            .tobytes()
+        ):
+            return False
+        # check if insertion runs bad
+        elif (
+            b"\x01" * gap_run
+            in numpy.logical_and(template_gaps, numpy.logical_not(seq_gaps))
+            .astype(numpy.uint8)
+            .tobytes()
+        ):
+            return False
+        return True
+
+    return result
+
+
+class _IndexableSeqs:
+    def __init__(self, parent):
+        self.parent = parent
+
+    def __getitem__(self, key):
+        return self.parent._make_aligned(key)
 
 
 class Alignment(SequenceCollection):
     def __init__(
         self,
-        slice_record: new_sequence.SliceRecordABC = None,
+        slice_record: new_sequence.SliceRecord = None,
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self._seqs_data.make_aligned = self._make_aligned
-        self._seqs_data.make_seq = self.moltype.make_seq
         self._slice_record = (
             slice_record
             if slice_record is not None
-            else SliceRecord(parent_len=self._seqs_data.align_len)
+            else new_sequence.SliceRecord(parent_len=self._seqs_data.align_len)
+        )
+
+    @singledispatchmethod
+    def __getitem__(self, index):
+        raise NotImplementedError(f"__getitem__ not implemented for {type(index)}")
+
+    @__getitem__.register
+    def _(self, index: str):
+        return self._make_aligned(seqid=index)
+
+    @__getitem__.register
+    def _(self, index: int):
+        new_slice = self._slice_record[index]
+        return self.__class__(
+            seqs_data=self._seqs_data,
+            slice_record=new_slice,
+            moltype=self.moltype,
+            info=self.info,
+        )
+
+    @__getitem__.register
+    def _(self, index: slice):
+        new_slice = self._slice_record[index]
+        return self.__class__(
+            seqs_data=self._seqs_data,
+            slice_record=new_slice,
+            moltype=self.moltype,
+            info=self.info,
+        )
+
+    def __repr__(self):
+        seqs = []
+        limit = 10
+        delimiter = ""
+        for count, name in enumerate(self.names):
+            if count == 3:
+                seqs.append("...")
+                break
+            elts = list(str(self.seqs[name])[: limit + 1])
+            if len(elts) > limit:
+                elts[-1] = "..."
+            seqs.append(f"{name}[{delimiter.join(elts)}]")
+        seqs = ", ".join(seqs)
+
+        return f"{len(self.names)} x {len(self)} {self.moltype.label} alignment: {seqs}"
+
+    def __len__(self):
+        return len(self._seqs_data)
+
+    def _make_aligned(self, seqid: str) -> Aligned:
+        # we set the slice record on the AlignedDataView
+        data = self._seqs_data.get_view(seqid)
+        data.slice_record = self._slice_record
+        return Aligned(data=data, moltype=self.moltype)
+
+    @property
+    def seqs(self) -> AlignedSeqsDataABC:
```

Comment:
Add a comment about redesign. We will make `_seqs` a private attribute and the indexable seqs class will be assigned to it in the constructor. The property will then just return the `_seqs` attribute.